### PR TITLE
feat(kit): `PdfViewer` add new component

### PR DESCRIPTION
**Pull request by waterplea**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [ ] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [ ] Feature
-   [ ] Refactoring
-   [ ] Code style update
-   [ ] Build or CI related changes
-   [ ] Documentation content changes

## What is the current behavior?

Closes # 

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [ ] No

### CODE REVIEW

File: projects/demo/src/modules/components/pdf-viewer/examples/2/index.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        @Inject(TuiPdfViewerService) private readonly pdfService: TuiPdfViewerService,
+    ) {}
+
+    show() {
+        const options: TuiPdfViewerOptions<Buttons> = {
+            label: 'Taiga UI',
+            actions: new PolymorpheusComponent(ActionsContent),
+            data: [
+                {
+                    text: 'Подписать',
```
Comment (splincode) [inline projects/demo/src/modules/components/pdf-viewer/examples/2/index.ts:36]: `Подписать` i18n?

File: projects/kit/components/pdf-viewer/pdf-viewer.component.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    selector: 'tui-pdf-viewer',
+    templateUrl: './pdf-viewer.template.html',
+    styleUrls: ['./pdf-viewer.style.less'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    animations: [tuiSlideInTop, tuiFadeIn],
+})
+export class TuiPdfViewerComponent<I, O> {
+    @HostBinding('@tuiSlideInTop')
+    @HostBinding('@tuiFadeIn')
+    readonly animation = {
```
Comment (splincode) [inline projects/kit/components/pdf-viewer/pdf-viewer.component.ts:19]: what do you think?

```
readonly animation = { value: '', ...this.options } as const;
```

File: projects/kit/components/pdf-viewer/pdf-viewer.service.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import {Injectable} from '@angular/core';
+import {AbstractTuiDialogService} from '@taiga-ui/cdk';
+import {PolymorpheusComponent} from '@tinkoff/ng-polymorpheus';
+import {TuiPdfViewerOptions} from './pdf-viewer-options';
+import {TuiPdfViewerComponent} from './pdf-viewer.component';
+
+const DIALOG = new PolymorpheusComponent(TuiPdfViewerComponent);
+const DEFAULT_OPTIONS = {
```
Comment (splincode) [inline projects/kit/components/pdf-viewer/pdf-viewer.service.ts:8]: what do you think?

```ts
const DEFAULT_OPTIONS = { label: '', actions: '' } as const;
```

Comment (waterplea) [reply projects/kit/components/pdf-viewer/pdf-viewer.service.ts:8]: I thought prettier takes care of it

Comment (waterplea) [reply projects/kit/components/pdf-viewer/pdf-viewer.service.ts:8]: Anyway, let's not comment on such things. If prettier thinks it's ok — we think it's ok

File: projects/kit/components/pdf-viewer/pdf-viewer.template.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
+<header class="header">
+    <h2 class="title">{{context.label}}</h2>
+    <div
+        polymorpheus-outlet
+        class="actions"
```
Comment (vladimirpotekhin) [inline projects/kit/components/pdf-viewer/pdf-viewer.template.html:5]: t- prefixes ?

File: projects/kit/components/pdf-viewer/pdf-viewer.template.html [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        [content]="context.actions"
+        [context]="context"
+    ></div>
+    <button
+        tuiIconButton
+        shape="rounded"
+        appearance=""
+        size="s"
+        icon="tuiIconCloseLarge"
+        title="Закрыть"
```
Comment (vladimirpotekhin) [inline projects/kit/components/pdf-viewer/pdf-viewer.template.html:15]: i18n

File: projects/cdk/abstract/dialog.directive.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import {ChangeDetectorRef, Directive, Inject, TemplateRef} from '@angular/core';
+import {TuiBaseDialogContext} from '@taiga-ui/cdk/interfaces';
+import {PolymorpheusTemplate} from '@tinkoff/ng-polymorpheus';
+import {AbstractTuiDialogService} from './dialog.service';
+import {EMPTY, Subject} from 'rxjs';
+import {endWith, ignoreElements, switchMap} from 'rxjs/operators';
+
+@Directive()
```
Comment (splincode) [inline projects/cdk/abstract/dialog.directive.ts:8]: what do you think?

```ts
type DialogContext<T> = TuiBaseDialogContext<void> & T;

export abstract class AbstractTuiDialogDirective<T> extends PolymorpheusTemplate<DialogContext<T>> {
 
constructor(
        @Inject(TemplateRef)
        templateRef: TemplateRef<DialogContext<T>>,
     // ...
}
```

File: projects/demo/src/modules/components/pdf-viewer/examples/2/index.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+                },
+                {
+                    text: 'Deny',
+                    onClick: context => context.completeWith('Document denied'),
+                },
+            ],
+        };
+
+        this.pdfService
+            .open<string>(new PolymorpheusComponent(PdfContent), options)
```
Comment (splincode) [inline projects/demo/src/modules/components/pdf-viewer/examples/2/index.ts:47]: why not encapsulate this behavior internally?

```ts
this.pdfService.open<string>(PdfContent, options)
```

for easy usage

File: projects/demo/src/modules/components/pdf-viewer/pdf-viewer.template.html [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+    <ng-template pageTab="Setup">
+        <ol class="b-demo-steps">
+            <li>
+                Import
+                <code>TuiPdfViewerModule</code>
+                to your main app module:
+
+                <tui-doc-code
+                    filename="myComponent.module.ts"
```
Comment (splincode) [inline projects/demo/src/modules/components/pdf-viewer/pdf-viewer.template.html:34]: `filename="my-component.module.ts"`

File: projects/demo/src/modules/components/pdf-viewer/pdf-viewer.template.html [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+                    [code]="exampleImportModule"
+                ></tui-doc-code>
+            </li>
+
+            <li>
+                Show preview with a service:
+
+                <tui-doc-code
+                    filename="myComponent.component.ts"
```
Comment (splincode) [inline projects/demo/src/modules/components/pdf-viewer/pdf-viewer.template.html:43]: `filename="my.component.ts"`

File: projects/kit/components/pdf-viewer/pdf-viewer.template.html (hunk 5 above)
Comment (splincode) [inline projects/kit/components/pdf-viewer/pdf-viewer.template.html:15]: i18n?

File: projects/kit/components/pdf-viewer/pdf-viewer.style.less [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    padding: 0 1rem 0 1.5625rem;
+    box-shadow: inset 0 -1px #535659;
+}
+
+.title {
+    margin: 0;
+    font: var(--tui-font-text-m);
+}
+
+.actions {
```
Comment (splincode) [inline projects/kit/components/pdf-viewer/pdf-viewer.style.less:25]: `t-*` prefix?

File: projects/demo/src/modules/components/pdf-viewer/examples/2/index.ts (hunk 7 above)
Comment (waterplea) [reply projects/demo/src/modules/components/pdf-viewer/examples/2/index.ts:47]: Because it works with PolymorpheusContent like everything else.

**Review by splincode — APPROVED** [review]

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**